Subbus/Classes/Subbus.swift

```diff
@@ -8,141 +8,241 @@
+        let eventType = String(reflecting: T.self)
+
+        subscriptions.removeAll(where: { $0.matches(identifier: id, eventType: eventType) })
+
+        log("Unegistered listener for identifier \"\(id)\" event \"\(eventType)\"")
+    }
+
     public static func unsubscribe<I>(id: I) {
-        //Verify data
-        guard (id as? OptionalProtocol) == nil else { log("Subscribe - You may not pass in an optional ID."); return }
-        
-        //Define strings
-        guard let identifier = stringFor(id: id) else { log("Unsubscribe - String representation of ID is empty."); return }
-        
-        //Retrieve Target Listeners
-        let listenerArray = shared.listeners.filter(filter(identifier: identifier))
-        
-        //Unsubscribe Listeners
-        for listener in listenerArray {
-            shared.eventbusCenter.removeObserver(listener.listener)
-        }
+        subscriptions.removeAll(where: { $0.matches(identifier: id) })
         
-        //Remove Listeners From Cache
-        while let index = shared.listeners.index(where: filter(identifier: identifier)) { //TODO: Maybe do structs; then you can probably do == comparison instead of filter as it stands
-            shared.listeners.remove(at: index)
-        }
-        
-        //Log
-        log("Unregistered all listeners for identifier \(identifier)")
+        log("Unegistered listener for identifier \"\(id)\"")
     }
     
-    fileprivate static func filter(identifier: String, eventName: String? = nil) -> (ListenerWrapper) -> Bool {
-        return {($0.identifier == identifier) && (eventName != nil ? $0.eventName == eventName : true)}
-    }
-    
-    //Misc
-    static func log(_ message: String) {
-        guard loggingEnabled == true else { return }
-        print("Subbus: \(message)")
+    public static func unsubscribe<T>(event: T.Type) {
+        let eventType = String(reflecting: T.self)
+        subscriptions.removeAll(where: { $0.matches(eventType: eventType) })
     }
-    
-    static func stringFor<I>(id: I) -> String? {
+}
+
+extension Subbus {
+    internal static func stringFor(id: Any) -> String? {
         var valString = "" //Value or pointer
-        
+        let idType = type(of: id)
+
         //Get the right string type; objects need memory address and primitives just need value.
-        if I.self is AnyClass {
+        if idType is AnyClass {
             valString = "\(Unmanaged.passUnretained(id as AnyObject).toOpaque())"
         } else {
             valString = "\(id)"
         }
-        
+
         //Ensure that the idString is valid
         guard valString.isEmpty == false else { return nil }
         guard valString != "nil" else { return nil }
+
+        return "\(idType)-\(valString)"
+    }
+    
+    internal static func log(_ message: String, force: Bool = false) {
+        guard logToConsole || force else { return }
+        print("Subbus: \(message)")
+    }
+    
+    internal static func cleanup() {
```

Comment:
This function name is vague. Rename to `removeAllSubscriptions()` or equivalent.

Reply:
Renamed to `clearInvalidSubscriptions`